src/solvers/dg_curve_quad/element_geometry.jl

```diff
@@ -0,0 +1,110 @@
+
+using StaticArrays: MArray
+
+mutable struct ElementGeometry{RealT<:Real, NNODES}
+  x       ::MArray{Tuple{NNODES, NNODES}, RealT}
+  y       ::MArray{Tuple{NNODES, NNODES}, RealT}
+  X_xi    ::MArray{Tuple{NNODES, NNODES}, RealT}
+  X_eta   ::MArray{Tuple{NNODES, NNODES}, RealT}
+  Y_xi    ::MArray{Tuple{NNODES, NNODES}, RealT}
+  Y_eta   ::MArray{Tuple{NNODES, NNODES}, RealT}
+  Jac     ::MArray{Tuple{NNODES, NNODES}, RealT}
+  invJac  ::MArray{Tuple{NNODES, NNODES}, RealT}
+  x_bndy  ::MArray{Tuple{NNODES, 4}, RealT}
+  y_bndy  ::MArray{Tuple{NNODES, 4}, RealT}
+  n_vecs  ::MArray{Tuple{NNODES, 4, 2}, RealT}
+  scaling ::MArray{Tuple{NNODES, 4}, RealT}
+end
```

Comment:
These types are not fully specified. For example,
```
julia> include("src/solvers/dg_curve_quad/test_mesh.jl")
[....]

julia> @code_warntype geometry_test.x
Variables
  #self#::Core.Const(getproperty)
  x::ElementGeometry{Float64, 12}
  f::Symbol

Body::MArray{S, Float64, N, L} where {S<:Tuple, N, L}
1 ─ %1 = Base.getfield(x, f)::MArray{S, Float64, N, L} where {S<:Tuple, N, L}
└──      return %1
```
Hence, you have basically two options to get performant (and type-stable) code:

- Introduce additional type parameters to make all `MArray`s fully typed.
- Make the types of the member parameters that are subtypes of `AbstractArray{T, N}` with appropriate choices of element type `T` and dimension `N`.

Feel free to ask me when it's not clear to you how to do that.

Reply:
Do you really want to have a `mutable struct` here or is a `struct` (immutable) also okay?

Reply:
Okay I am now reading more about the typing. But I am slightly confused because I thought (for example) that doing something like `scaling ::MArray{Tuple{NNODES, 4}, RealT}` would tell Julia that it is type real.

Reply:
Yes, it is - but it's still not fully specified, since `MArray`s have more type parameters:
```julia
julia> using StaticArrays

julia> MArray{Tuple{5, 4}, Float64}
MArray{Tuple{5, 4}, Float64, N, L} where {N, L}
```
Note the `where {N, L}`.

Reply:
Aha, gotcha. So I can get away from using the `RealT`?

Reply:
`RealT` is fine as you use it. What I meant is that you should 

- either add additional type parameters to specify the remaining ones of `MArray` completely (see the `where` clause in the example above)
- or make the types of `x` etc. type paremeters, e.g. as `DerivativeMatrix` in `LobattoLegendreBasis`(https://github.com/trixi-framework/Trixi.jl/blob/main/src/solvers/dg/basis_lobatto_legendre.jl#L7-L22). 

You can also just ignore that part for now, make the basic algorithms correct using the current typing, and ping me later - we could also look at it together in the next hackfest. Fixing the type instability is more a matter of "make it fast" while our usual development workflow is
1. Make it work
2. Make it fast